pkg/proxy/ipvs/ipset.go

```diff
@@ -112,76 +135,84 @@ func NewIPSet(handle utilipset.Interface, name string, setType utilipset.Type, i
 			}
 		}
 	}
-	set := &IPSet{
-		IPSet: utilipset.IPSet{
-			Name:       name,
-			SetType:    setType,
-			HashFamily: hashFamily,
-			Comment:    comment,
-		},
-		activeEntries: sets.NewString(),
-		handle:        handle,
+	set := &utilipset.IPSet{
+		Name:       name,
+		SetType:    setType,
+		HashFamily: hashFamily,
+		Comment:    comment,
 	}
+	set.SetIPSetDefaults()
 	return set
 }
 
-func (set *IPSet) validateEntry(entry *utilipset.Entry) bool {
-	return entry.Validate(&set.IPSet)
+func (s *ipsetManager) isSetEmpty(setName string) bool {
+	return s.activeEntries[setName].Len() == 0
 }
 
-func (set *IPSet) isEmpty() bool {
-	return len(set.activeEntries.UnsortedList()) == 0
+func (s *ipsetManager) getSetComment(setName string) string {
+	return fmt.Sprintf("\"%s\"", s.setList[setName].Comment)
 }
 
-func (set *IPSet) getComment() string {
-	return fmt.Sprintf("\"%s\"", set.Comment)
+func (s *ipsetManager) resetEntries() {
+	for setName := range s.activeEntries {
+		s.activeEntries[setName] = sets.NewString()
+	}
 }
 
-func (set *IPSet) resetEntries() {
-	set.activeEntries = sets.NewString()
+func (s *ipsetManager) entryValid(setName string, entry *utilipset.Entry) bool {
+	if s.setList[setName] == nil {
+		return false
+	}
+	return entry.Validate(s.setList[setName])
 }
 
-func (set *IPSet) syncIPSetEntries() {
-	appliedEntries, err := set.handle.ListEntries(set.Name)
-	if err != nil {
-		klog.Errorf("Failed to list ip set entries, error: %v", err)
+func (s *ipsetManager) insertIPSetEntry(setName string, entry *utilipset.Entry) {
+	if !s.entryValid(setName, entry) {
+		klog.Errorf("%s", fmt.Sprintf(EntryInvalidErr, entry, setName))
```

Comment:
Instead of logging here, would prefer `insertIPSetEntry` return an error and log at proxier level, what do you think?

Reply:
Normally this `Validate` should not return err, and if this do happened, I think this should not block the whole syncRule work, so I think logging here or at proxier level doesn't metter. Is there anything I haven't thought about?

Reply:
Logging at the proxier level is better for visibility since you'll know at what point of `syncProxyRules` the error is happening. If you log at the insert level, all insert failures log at the same place.

Reply:
Good point, thanks a lot. : )